src/WCSimPhysicsList.cc

```diff
@@ -392,349 +369,341 @@ void WCSimPhysicsList::ConstructGeneral()
+	  theInelasticProcess->RegisterMe(theBertiniModel);
+
+	  // Added FTFP Model - SS
+	  FTFP_model->SetMinEnergy(3.0*CLHEP::GeV);
+	  theInelasticProcess->RegisterMe(FTFP_model);
+
+ 	  pmanager->AddDiscreteProcess(theInelasticProcess);
+ 	}
+ 
+      else if (particleName == "kaon-")
+ 	{
+ 	  pmanager->AddDiscreteProcess(theElasticProcess);
+	  G4KaonMinusInelasticProcess* theInelasticProcess =
+ 	    new G4KaonMinusInelasticProcess();
+
+	  // Added Bertini Model - FDL
+	  G4CascadeInterface* theBertiniModel = new G4CascadeInterface;
+	  theBertiniModel->SetMaxEnergy(5.0*CLHEP::GeV);
+	  theInelasticProcess->RegisterMe(theBertiniModel);
+
+	  // Added FTFP Model - SS
+	  FTFP_model->SetMinEnergy(4.0*CLHEP::GeV);
+	  theInelasticProcess->RegisterMe(FTFP_model);
+
+ 	  pmanager->AddDiscreteProcess(theInelasticProcess);
+	  pmanager->AddRestProcess(new G4KaonMinusAbsorptionBertini, ordDefault);
+ 	}
+ 
+      else if (particleName == "proton")
 	{
 	  pmanager->AddDiscreteProcess(theElasticProcess);
-	  G4ProtonInelasticProcess* theInelasticProcess = 
+	  G4ProtonInelasticProcess* theInelasticProcess =
 	    new G4ProtonInelasticProcess();
 
-	  
-	  //=================================
-	  // Added by JLR 2005-07-05
-	  //=================================
-	  // Options for secondary interaction models
-	  // Choice defined in jobOptions.mac, which is
-	  // read in before initialization of the run manager.
-	  // In the absence of this file, BINARY will be used.
-	  // Gheisha = Original Geant4 default 
-	  // Bertini = Bertini intra-nuclear cascade model
-	  // Binary  = Binary intra-nuclear cascade model
-	  if (gheishahad) {
-	    G4LEProtonInelastic* theLEInelasticModel = new G4LEProtonInelastic;
-	    theInelasticProcess->RegisterMe(theLEInelasticModel);
-	  }
-	  else if (bertinihad) {
+	  if (bertinihad) {
 	    G4CascadeInterface* theBertiniModel = new G4CascadeInterface;
+	    theBertiniModel->SetMaxEnergy(5.0*CLHEP::GeV);
 	    theInelasticProcess->RegisterMe(theBertiniModel);
 	  }
 	  else if (binaryhad) {
 	    G4BinaryCascade* theBinaryModel = new G4BinaryCascade();
+	    theBinaryModel->SetMaxEnergy(5.0*CLHEP::GeV);
 	    theInelasticProcess->RegisterMe(theBinaryModel);
-
-	    G4LEProtonInelastic* theLEInelasticModel = new G4LEProtonInelastic;
-	    theLEInelasticModel->SetMinEnergy(10.1*GeV);
-	    theLEInelasticModel->SetMaxEnergy( 45.*GeV );
-	    theInelasticProcess->RegisterMe(theLEInelasticModel);
-
 	  }
 	  else {
-	    G4cout << "No secondary interaction model chosen! Using G4 BINARY." << G4endl;
-	    G4BinaryCascade* theBinaryModel = new G4BinaryCascade();
-	    theInelasticProcess->RegisterMe(theBinaryModel);
+	    G4cout << "No secondary interaction model chosen! Using G4 BERTINI." << G4endl;
+	    G4CascadeInterface* theBertiniModel = new G4CascadeInterface;
+	    theBertiniModel->SetMaxEnergy(5.0*CLHEP::GeV);
+	    theInelasticProcess->RegisterMe(theBertiniModel);
 	  }
```

Comment:
This code changes the default for the secondary interactions to bertini and FTFP (rather than binary), but it hasn't been changed in jobOptions2.mac. Is the proposal to change the secondary interactions model? If so, we should change it in the macro. Again, @fdilodov, you did some work on this. Do you know if switching the secondary model is the intention here?